### ISSUE TITLE
Ingress depreciation warning does not seem to honor   --providers.kubernetesingress.ingressclass settings

### ISSUE THREAD

**Issue by SimonTheLeg** [containous/traefik#6746]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

Start Traefik inside kubernetes with the following CMD flags

```
  - --providers.kubernetescrd=false # we use native k8s ingress instead of the crd
  - --providers.kubernetesingress
  - --providers.kubernetesingress.ingressclass=traefik
  - --log.level=debug
  - --entrypoints.web.http.redirections.entrypoint.to=:443
  - --entrypoints.web.http.redirections.entrypoint.permanent=true
```

### What did you expect to see?

No depreciation warning for ingress that does not have the ingressclass traefik annotation, as traefik should ignore this ingress.

the ingress in question
```bash
kc get ing -n default dev-ingress -o yaml | grep traefik
# returns nothing
```

### What did you see instead?

```log
time="2020-04-29T10:22:18Z" level=warning msg="Ingress default/dev-ingress: the apiVersion 'extensions/v1beta1' is deprecated, use 'networking.k8s.io/v1beta1' instead."
```

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      2.2.0
Codename:     chevrotin
Go version:   go1.14.1
Built:        2020-03-25T17:32:57Z
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

see above

### If applicable, please paste the log output in DEBUG level (`--log.level=DEBUG` switch)

surrounding logs:
```
time="2020-04-29T10:22:18Z" level=warning msg="Ingress default/dev-ingress: the apiVersion 'extensions/v1beta1' is deprecated, use 'networking.k8s.io/v1beta1' instead."
time="2020-04-29T10:22:18Z" level=warning msg="Ingress monitoring/grafana: the apiVersion 'extensions/v1beta1' is deprecated, use 'networking.k8s.io/v1beta1' instead."
time="2020-04-29T10:22:18Z" level=warning msg="Ingress monitoring/pushprox-endpoint: the apiVersion 'extensions/v1beta1' is deprecated, use 'networking.k8s.io/v1beta1' instead."
time="2020-04-29T10:22:18Z" level=debug msg="Skipping Kubernetes event kind *v1.Endpoints" providerName=kubernetes
time="2020-04-29T10:22:18Z" level=debug msg="Serving default certificate for request: \"pushprox.cluster.dev.meshcloud.io\""
time="2020-04-29T10:22:18Z" level=debug msg="http: TLS handshake error from 10.156.0.22:16018: remote error: tls: bad certificate"
time="2020-04-29T10:22:18Z" level=warning msg="Ingress default/dev-ingress: the apiVersion 'extensions/v1beta1' is deprecated, use 'networking.k8s.io/v1beta1' instead."
time="2020-04-29T10:22:18Z" level=warning msg="Ingress monitoring/grafana: the apiVersion 'extensions/v1beta1' is deprecated, use 'networking.k8s.io/v1beta1' instead."
time="2020-04-29T10:22:18Z" level=warning msg="Ingress monitoring/pushprox-endpoint: the apiVersion 'extensions/v1beta1' is deprecated, use 'networking.k8s.io/v1beta1' instead."
time="2020-04-29T10:22:18Z" level=debug msg="Skipping Kubernetes event kind *v1.Endpoints" providerName=kubernetes
time="2020-04-29T10:22:19Z" level=warning msg="Ingress monitoring/pushprox-endpoint: the apiVersion 'extensions/v1beta1' is deprecated, use 'networking.k8s.io/v1beta1' instead."
time="2020-04-29T10:22:19Z" level=warning msg="Ingress default/dev-ingress: the apiVersion 'extensions/v1beta1' is deprecated, use 'networking.k8s.io/v1beta1' instead."
time="2020-04-29T10:22:19Z" level=warning msg="Ingress monitoring/grafana: the apiVersion 'extensions/v1beta1' is deprecated, use 'networking.k8s.io/v1beta1' instead."
time="2020-04-29T10:22:19Z" level=debug msg="Skipping Kubernetes event kind *v1.Endpoints" providerName=kubernetes
time="2020-04-29T10:22:19Z" level=warning msg="Ingress default/dev-ingress: the apiVersion 'extensions/v1beta1' is deprecated, use 'networking.k8s.io/v1beta1' instead."
time="2020-04-29T10:22:19Z" level=warning msg="Ingress monitoring/grafana: the apiVersion 'extensions/v1beta1' is deprecated, use 'networking.k8s.io/v1beta1' instead."
time="2020-04-29T10:22:19Z" level=warning msg="Ingress monitoring/pushprox-endpoint: the apiVersion 'extensions/v1beta1' is deprecated, use 'networking.k8s.io/v1beta1' instead."
time="2020-04-29T10:22:19Z" level=debug msg="Skipping Kubernetes event kind *v1.Endpoints" providerName=kubernetes
time="2020-04-29T10:22:20Z" level=warning msg="Ingress default/dev-ingress: the apiVersion 'extensions/v1beta1' is deprecated, use 'networking.k8s.io/v1beta1' instead."
time="2020-04-29T10:22:20Z" level=warning msg="Ingress monitoring/grafana: the apiVersion 'extensions/v1beta1' is deprecated, use 'networking.k8s.io/v1beta1' instead."
time="2020-04-29T10:22:20Z" level=warning msg="Ingress monitoring/pushprox-endpoint: the apiVersion 'extensions/v1beta1' is deprecated, use 'networking.k8s.io/v1beta1' instead."
time="2020-04-29T10:22:20Z" level=debug msg="Skipping Kubernetes event kind *v1.Endpoints" providerName=kubernetes
time="2020-04-29T10:22:20Z" level=warning msg="Ingress monitoring/pushprox-endpoint: the apiVersion 'extensions/v1beta1' is deprecated, use 'networking.k8s.io/v1beta1' instead."
time="2020-04-29T10:22:20Z" level=warning msg="Ingress default/dev-ingress: the apiVersion 'extensions/v1beta1' is deprecated, use 'networking.k8s.io/v1beta1' instead."
time="2020-04-29T10:22:20Z" level=warning msg="Ingress monitoring/grafana: the apiVersion 'extensions/v1beta1' is deprecated, use 'networking.k8s.io/v1beta1' instead."
time="2020-04-29T10:22:20Z" level=debug msg="Skipping Kubernetes event kind *v1.Endpoints" providerName=kubernetes
time="2020-04-29T10:22:20Z" level=warning msg="Ingress monitoring/pushprox-endpoint: the apiVersion 'extensions/v1beta1' is deprecated, use 'networking.k8s.io/v1beta1' instead."
time="2020-04-29T10:22:20Z" level=warning msg="Ingress default/dev-ingress: the apiVersion 'extensions/v1beta1' is deprecated, use 'networking.k8s.io/v1beta1' instead."
time="2020-04-29T10:22:20Z" level=warning msg="Ingress monitoring/grafana: the apiVersion 'extensions/v1beta1' is deprecated, use 'networking.k8s.io/v1beta1' instead."
time="2020-04-29T10:22:20Z" level=debug msg="Skipping Kubernetes event kind *v1.Endpoints" providerName=kubernetes
```

At this point I am also wondering: **Is it intended that traefik spams these log lines every couple seconds?** It makes it kind of hard to read the rest of the log

**Comment by ldez** [comment]
Duplicate of #6536

**Comment by SimonTheLeg** [comment]
This is only a partial duplicate in my opinion the "Is it intended that traefik spams these log lines every couple seconds?" is definitely a duplicate. 

But the original question still remains, why is traefik checking ingress depreciation on ingresses that don't match the class annotation?

**Comment by ldez** [comment]
The problem is fixed by #6624.

In fact it's an issue with k8s api, not really with Traefik.
